static/src/javascripts/projects/common/modules/identity/email-preferences.js

```diff
@@ -0,0 +1,122 @@
+// This is a workaround for the email preferences page https://profile.thegulocal.com/email-prefs
+// We want to submit subscribe/unsubscribe requests without a full page refresh
+// Hopefully this will be short-lived; if it is still alive in 2017, git blame and cry
+
+define([
+    'bean',
+    'qwery',
+    'reqwest',
+    'fastdom',
+    'common/utils/$'
+], function (
+    bean,
+    qwery,
+    reqwest,
+    fastdom,
+    $
+) {
+    function useReqwest(buttonEl) {
+        bean.on(buttonEl, 'click', function () {
+            buttonEl.disabled = true;
+            buttonEl.innerHTML = 'Loading...';
+            var formQueryString = generateFormQueryString(buttonEl);
+            reqwest({
+                url: '/email-prefs',
+                method: 'POST',
+                data: formQueryString,
+                error: function () {
+                    renderErrorMessage(buttonEl);
+                },
+                success: function (response) {
+                    var subscriptionState;
+                    try {
+                        // Why array.length rather than typeof; there are multiple circumstances where the value could be undefined:
+                        // nothing came back at all
+                        // stuff came back but that one field is undefined
+                        // the data came back in the format expected as above, but with the subscribeTo value undefined
+                        if (response.subscriptions.length < 1) {
+                            subscriptionState = false;
+                        } else {
+                            subscriptionState = response.subscriptions[0].subscribedTo;
+                        }
+                    } catch (err) {
+                        renderErrorMessage(buttonEl);
+                    } finally {
+                        updateButton(buttonEl, subscriptionState);
+                    }
+                }
+            });
+        });
+    }
+
+    function enhanceEmailPreferences() {
+        $.forEachElement('.email-subscription__button', useReqwest);
+    }
+
+    function encodeFormData(csrfToken, buttonVal, htmlPreference) {
+        return 'csrfToken=' + encodeURIComponent(csrfToken) + '&' +
+        'addEmailSubscription=' + encodeURIComponent(buttonVal) + '&' +
+        'htmlPreference=' + encodeURIComponent(htmlPreference);
+    }
+
+    function renderErrorMessage(buttonEl) {
+        // appends an error message on the parent div of the button
+        var errorMessage = 'Sorry, an error has occurred, please refresh the page and try again';
+        return fastdom.write(function () {
+            var insertionPoint = $.ancestor(buttonEl, 'email-subscription');
+            // Only append an error message once for each email subscription DIV
+            if (qwery('.form__error', insertionPoint).length < 1) {
```

Comment:
a better approach might be to remove any error messages before sending a new request? otherwise I think you could end up with stale error messages against other buttons